### PR TITLE
util: allow to run criu under strace

### PR DESCRIPTION
**Pull request by avagin**
fork_and_ptrace_attach has to fork a child with CLONE_UNTRACED, so that strace doesn't trace it.

### CODE REVIEW

File: criu/util.c [hunk 1]
```diff
@@ -661,40 +661,53 @@ int cr_system_userns(int in, int out, int err, char *cmd, char *const argv[], un
 	return ret;
 }
 
+struct child_args {
+	int *sk_pair;
+	int (*child_setup)(void);
+};
+
+static int child_func(void *_args) {
```
Comment (rst0git) [inline criu/util.c:669]:
```diff
-static int child_func(void *_args) {
+static int child_func(void *_args)
+{
```

**Comment by rst0git** [comment]
LGTM, but I was not able to replicate the problem in https://github.com/checkpoint-restore/criu/issues/2269#issuecomment-1739927149.

**Comment by avagin** [comment]
@rst0git it is triggered each time when kerndat_has_ptrace_get_rseq_conf is called. If you can't reproduce it, you need to find out why kerndat_has_ptrace_get_rseq_conf isn't executed in your case.

https://github.com/checkpoint-restore/criu/blob/5bf7652504530b9baa04f411052f9c079c7f14eb/criu/kerndat.c#L978

**Comment by rst0git** [comment]
@avagin Thanks! I was able to replicate it and confirm that it works.

**Review by rst0git — APPROVED** [review]

**Pull request merged**